external-crates/move/move-binary-format/src/deserializer.rs

```diff
@@ -35,23 +35,34 @@ impl CompiledScript {
 impl CompiledModule {
     /// Deserialize a &[u8] slice into a `CompiledModule` instance.
     pub fn deserialize(binary: &[u8]) -> BinaryLoaderResult<Self> {
-        Self::deserialize_with_max_version(binary, VERSION_MAX)
+        Self::deserialize_with_config(
+            binary,
+            VERSION_MAX,
+            /* check_no_extraneous_bytes */ false,
+        )
     }
 
-    /// Deserialize a &[u8] slice into a `CompiledModule` instance, up to the specified version.
-    pub fn deserialize_with_max_version(
+    /// Deserialize a &[u8] slice into a `CompiledModule` instance with settings
+    /// - Can specify up to the specified version.
+    /// - Can specify if the deserializer should error on trailing bytes
+    pub fn deserialize_with_config(
```

Comment:
this is just for another day but I'd think we should rename `deserialize` to `deserialize_with_defaults` and change `deserialize_with_config` to `deserialize`